agent/xds/listeners_ingress.go

```diff
@@ -19,6 +19,15 @@ func (s *ResourceGenerator) makeIngressGatewayListeners(address string, cfgSnap
 	for listenerKey, upstreams := range cfgSnap.IngressGateway.Upstreams {
 		var tlsContext *envoy_tls_v3.DownstreamTlsContext
 
+		listenerCfg, ok := cfgSnap.IngressGateway.Listeners[listenerKey]
```

Comment:
Why fetch this here and then all over again in `resolveListenerSDSConfig` when you could just pass it down through?

Reply:
I did start that refactor but I ended up backing out as it ended up changing quite a few lines that reference `listenerKey` all just to save one map lookup... It doesn't allow us to remove the `error` return which was my initial motivation because there are other possible error sources too.

I can do that though if you think the code is significantly better that way?